app/scripts/modules/azure/serverGroup/details/rollback/RollbackServerGroupModal.tsx

```diff
@@ -0,0 +1,162 @@
+import * as React from 'react';
+
+import Select from 'react-select';
+
+import {
+  IModalComponentProps,
+  noop,
+  NgReact,
+  ReactInjector,
+  ReactModal,
+  TaskMonitor,
+  TaskReason,
+} from '@spinnaker/core';
+
+import { AzureModalFooter } from '../../../common/AzureModalFooter';
+
+export interface IAzureRollbackServerGroupModalProps extends IModalComponentProps {
+  application: any;
+  serverGroup: any;
+  disabledServerGroups: any;
+}
+
+export interface IAzureRollbackServerGroupModalState {
+  taskMonitor: TaskMonitor;
+  submitting: boolean;
+  command: any;
+}
+
+export class AzureRollbackServerGroupModal extends React.Component<
+  IAzureRollbackServerGroupModalProps,
+  IAzureRollbackServerGroupModalState
+> {
+  public static defaultProps: Partial<IAzureRollbackServerGroupModalProps> = {
+    closeModal: noop,
+    dismissModal: noop,
+  };
+
+  public static show(props: IAzureRollbackServerGroupModalProps) {
+    const modalProps = {};
+    return ReactModal.show(AzureRollbackServerGroupModal, props, modalProps);
+  }
+
+  constructor(props: IAzureRollbackServerGroupModalProps) {
+    super(props);
+
+    const { application, serverGroup } = props;
+
+    this.state = {
+      taskMonitor: new TaskMonitor({
+        application: application,
+        title: 'Rollbacking your server group',
+        modalInstance: TaskMonitor.modalInstanceEmulation(() => this.props.dismissModal()),
+      }),
+      submitting: true,
+      command: {
+        interestingHealthProviderNames: [],
+        rollbackType: 'EXPLICIT',
+        rollbackContext: {
+          rollbackServerGroupName: serverGroup.name,
+          enableAndDisableOnly: true,
+        },
+      },
+    };
+  }
+
+  private close = (args?: any) => {
+    this.props.dismissModal.apply(null, args);
+  };
+
+  private submit = () => {
+    const { command, taskMonitor } = this.state;
+    const { serverGroup, application } = this.props;
+
+    taskMonitor.submit(() => {
+      return ReactInjector.serverGroupWriter.rollbackServerGroup(serverGroup, application, command);
+    });
+  };
+
+  private filterServerGroups = (disabledServerGroups: any) => {
+    const filteredDisabledServerGroups = disabledServerGroups
+      .filter((disabledServerGroup: any) => disabledServerGroup.instanceCounts.total !== 0)
+      .sort((a: any, b: any) => b.name.localeCompare(a.name));
+
+    return filteredDisabledServerGroups;
+  };
+
+  private isValid = () => {
+    const restoreServerGroupName = this.state.command.rollbackContext.restoreServerGroupName;
+    return restoreServerGroupName !== undefined;
+  };
+
+  private handleServerGroupChange = (restoreServerGroupOption: any) => {
+    const { disabledServerGroups } = this.props;
+    const newCommand = this.state.command;
```

Comment:
It is more typical in React to make a copy of the object when making changes to state, like this:

```jsx
const newCommand = { ...this.state.command };
// make changes to newCommand
this.setState({
  command: newCommand
});
```

I believe this is for performance - it is faster for the library to check equality between two different objects than it is to check equality between all the primitive fields of two objects.  I'm not 100% sure how Deck approaches this however and might be wrong about this.